language/tools/move-unit-test/src/cargo_runner.rs

```diff
@@ -6,22 +6,29 @@ use crate::UnitTestingConfig;
 pub fn run_tests_with_config_and_filter(
     mut config: UnitTestingConfig,
     root_path: &str,
-    pattern: &str,
+    source_pattern: &str,
+    dep_root_and_pattern: Option<(&str, &str)>,
 ) {
-    let re = regex::Regex::new(pattern)
-        .unwrap_or_else(|_| panic!("Invalid regular expression: '{}'", pattern));
-    let sources = move_stdlib::utils::iterate_directory(&std::path::Path::new(root_path))
+    let get_files = |root_path, pat| {
+    let source_re = regex::Regex::new(pat)
+        .unwrap_or_else(|_| panic!("Invalid regular expression: '{}'", pat));
+    move_stdlib::utils::iterate_directory(&std::path::Path::new(root_path))
         .filter_map(|path| {
             let name = path.to_string_lossy();
-            if re.is_match(&name) {
+            if source_re.is_match(&name) {
                 Some(name.to_string())
             } else {
                 None
             }
         })
-        .collect();
+        .collect()
+    };
+
+    let sources = get_files(root_path, source_pattern);
```

Comment:
I'm having trouble locating this function, what does `get_files` do?

Reply:
`get_path` is on line 12. Basically it gets all of the files matching the regex in `source_pattern` that are under `root_path`, i.e., something along the lines of (in bash) `ls -R <root_path> <source_pattern>`

Reply:
ah I see it now. Is that really how Rust wants to indent that? strange

But at any rate, the pattern feels good for source files because you want to do something like "Run this subsection of tests" or something. But do we really want that for deps?

Reply:
Agree that the indentation is super weird here.

For deps this can make sense (IMO), since you may want to only include a subsection of tests, and also to filter out non `.move` files (although we could default this, although I would err in favor of giving the programmer more controls here as otherwise it could cause some pain).

Reply:
The main entrypoint to this function is through the `register_move_unit_tests!` macro. I'll add an additional option to that in there where you only need to specify the `dep_root` and the pattern will be defaulted to `r"*.move"`, but if you need it, you will still have the option of defining a specific pattern for dep files.

Reply:
This feels a bit inconsistent with the other tools where you pass multiple paths to directories or files, and the compiler finds all of the move/mv files. I don't see the value in complicating with with regexes

Reply:
Fair point -- we can always add that in later if we find we need it. I've removed the regex for deps for now